From: Dan Carpenter <dan.carpenter@oracle.com>
Date: Thu, 10 Aug 2017 12:54:59 +0300
Subject: hns3pf: fix hns3_del_tunnel_port()
Patch-mainline: v4.14-rc1
Git-commit: 9537e7cbf20787416b25833f2f9886166d8148f2
References: bsc#1104353 FATE#326415

This function has a copy and paste bug so it accidentally calls the add
function instead of the delete function.

Fixes: 76ad4f0ee747 ("net: hns3: Add support of HNS3 Ethernet Driver for hip08 SoC")
Signed-off-by: Dan Carpenter <dan.carpenter@oracle.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/net/ethernet/hisilicon/hns3/hns3pf/hns3_enet.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/ethernet/hisilicon/hns3/hns3pf/hns3_enet.c
+++ b/drivers/net/ethernet/hisilicon/hns3/hns3pf/hns3_enet.c
@@ -1138,7 +1138,7 @@ static void hns3_del_tunnel_port(struct
 	udp_tnl->dst_port = 0;
 	/* TBD send command to hardware to del port  */
 	if (h->ae_algo->ops->del_tunnel_udp)
-		h->ae_algo->ops->add_tunnel_udp(h, port);
+		h->ae_algo->ops->del_tunnel_udp(h, port);
 }
 
 /* hns3_nic_udp_tunnel_add - Get notifiacetion about UDP tunnel ports
